### ISSUE TITLE
ESM Support

### ISSUE THREAD

**Issue by dsanders11** [electron/fiddle#1466]
With ESM support coming to Electron 28, it would be nice to have support for `main.mjs` in Fiddle when that release happens. Rough sketch of what I think is necessary to support it.

- Add support for `.mjs` as a file extension
- Support either `main.js` or `main.mjs` for main process code
- Smooth out the UX
  - Don't allow both `main.js` and `main.mjs` to exist in a Fiddle
  - Consider whichever is present to be the required file
  - Show a warning to the user if they're trying to use `main.mjs` in a version before 28

Bonus points UX: Add a "Convert to ESM" button to the context menu when you right click `main.js`

**Comment by bnb** [comment]
> Add support for .mjs as a file extension

is .cjs also already supported? Worth doing if not already.

**Comment by dsanders11** [comment]
> is .cjs also already supported? Worth doing if not already.

Fair point. It is not, at the moment only `.js`, `.html`, and `.css` extensions are supported. Shouldn't be much extra work to toss in `.cjs` when refactoring to support alternatives to `main.js` as the main entrypoint.
